client-js/app/scripts/client_rpc.ts

```diff
@@ -0,0 +1,94 @@
+import promise = require('es6-promise');
+
+var Promise = promise.Promise;
+
+export function sendRequest(method: string, params: Object, socket, server: string): Promise<string> {
+        console.log("sending dumb RPC named: "+ method+" argument: "+ JSON.stringify(params));
+        return socket.call("org.labrad."+server+"." + method, params)
+                      .then((result) => <string>result);
+}
```

Comment:
indentation should be 2 spaces. This looks like tabs? Also, I think this function should be part of the `JsonRpcSocket` class, and the class can have a member called `prefix` that gets added to the front of the method name for every request that is sent. That way even "org.labrad" does not need to be hardcoded and we keep things maximally flexible, but still encapsulated within JsonRpcSocket.